parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputFormat.java

```diff
@@ -253,7 +283,23 @@ public ParquetInputFormat() {
   @Override
   public List<InputSplit> getSplits(JobContext jobContext) throws IOException {
     Configuration configuration = ContextUtil.getConfiguration(jobContext);
-    return new ArrayList<InputSplit>(getSplits(configuration, getFooters(jobContext)));
+    List<InputSplit> splits = new ArrayList<InputSplit>();
+
+    if (isTaskSideMetaData(configuration)) {
+      // Although not required by the API, some clients may depend on always
+      // receiving ParquetInputSplit. Translation is required at some point.
+      for (InputSplit split : super.getSplits(jobContext)) {
+        Preconditions.checkArgument(split instanceof FileSplit,
+            "Cannot wrap non-FileSplit: " + split);
+        splits.add(ParquetInputSplit.from((FileSplit) split));
```

Comment:
I wasn't talking about Hive, and you're right that Hive uses its own getSplits method.

I think it's a common practice to assume the type of the splits returned by getSplits, as we do here assuming that the FileInputFormat returns FileSplits. Originally, I had this return the FileSplit itself and handled both types in the RecordReader, but this would needlessly break any client that assumed ParquetInputSplit. This seemed unnecessary until we're ready to get rid of ParquetInputSplit, but that depends on deprecating the client-side behavior and is definitely a separate issue.